sklearn/metrics/tests/test_score_objects.py

```diff
@@ -500,18 +500,20 @@ def test_scorer_sample_weight():
         if name in REQUIRE_POSITIVE_Y_SCORERS:
             target = _require_positive_y(target)
         try:
-            weighted = scorer(estimator[name], X_test, target,
-                              sample_weight=sample_weight)
-            ignored = scorer(estimator[name], X_test[10:], target[10:])
-            unweighted = scorer(estimator[name], X_test, target)
-            assert weighted != unweighted, (
-                "scorer {0} behaves identically when "
-                "called with sample weights: {1} vs "
-                "{2}".format(name, weighted, unweighted))
-            assert_almost_equal(weighted, ignored,
-                                err_msg="scorer {0} behaves differently when "
-                                "ignoring samples and setting sample_weight to"
-                                " 0: {1} vs {2}".format(name, weighted,
+            if name != 'neg_median_absolute_error':
```

Comment:
can you help me undestand why you'd need this?

Also if we need to ignore `neg_median_absolute_error`, it would be preferable to filter it out before the loop `for name, scorer in SCORERS.items():`

Reply:
Since our data is binary classification data, with target being 1 or 0, there are only 2 possible values of `np.abs(y_pred - y_true)`, 1 or 0. With only 2 possible values it's likely that median after 'randomly' removing/weighting as zero 10 elements (out of 25) is the same as the median with all 25 elements.

To filter before the loop would you make a copy of `SCORER` and remove `neg_median_absolute_error` before the loop? Is there a better way?

Reply:
I could instead use `make_regression` to create a `y` just for `neg_median_absolute_error`?

Reply:
I think using regression targets here for regression metrics seems reasonable...?

TBH, I think this test is weird. Why are we testing the metric properties of scorers when really the unit that this file should be testing is the wrapper around the metric? We should only be testing here that `sample_weight` is correctly passed to the metric, regardless of the parameters to `make_scorer`.

Reply:
Fair point. Though I would like to add more tests for `_weighted_percentile` wrt to weighted median. Specifically, that `_weighted_percentile(percentile=50)` with equal weights is same as `np.median` and also that the difference between sum of weights left and right of the weighted median are the smallest possible. Reference: https://en.wikipedia.org/wiki/Weighted_median#Properties
If you agree, should I add the tests to this PR or a different one?

Also is `test_gradient_boosting_loss_functions.py` the best place for these tests? Makes it a bit difficult to find.

With regard to this PR, should I use regression targets just for `neg_median_absolute_error` or all regression metrics?

Reply:
> With regard to this PR, should I use regression targets just for neg_median_absolute_error or all regression metrics?

if it works for all, that would be more elegant.

> Also is test_gradient_boosting_loss_functions.py the best place for these tests? Makes it a bit difficult to find.

Not `sklearn/metrics/tests/test_regreession.py`?

Reply:
Whoops I meant the tests for `_weighted_percentile`, which are in `test_gradient_boosting_loss_functions.py`. I would like to add to test for `_weighted_percentile`

Reply:
starting here: https://github.com/scikit-learn/scikit-learn/blob/2d12bb57506d72a110eaedcf65783fd49a045854/sklearn/ensemble/tests/test_gradient_boosting_loss_functions.py#L106